src/crud.js

```diff
@@ -0,0 +1,185 @@
+import dots from '../assets/images/dots.png';
+import removeIcon from '../assets/images/delete.png';
+import saveImg from '../assets/images/save.jpeg';
+
+class Todo {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+const deleteTodos = (e) => {
+  const removeBtn = e.target;
+  const btnClass = removeBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  let existingTodos = JSON.parse(localStorage.getItem('todos'));
+  existingTodos = existingTodos.filter((todos, index) => index !== id);
+  removeBtn.parentNode.remove();
+  existingTodos.forEach((task, i) => (task.index = i + 1));
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+  createTodos();
+};
+
+const clearCompletedTodos = () => {
+  const completed = document.querySelectorAll('input:checked');
+  let existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (completed) {
+    completed.forEach((elem) => {
+      const task = elem.parentNode;
+      task.parentNode.remove();
+    });
+  }
+  existingTodos = existingTodos.filter((todos) => todos.completed === false);
+  existingTodos.forEach((task, i) => (task.index = i + 1));
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+};
+
+const saveTodos = (e) => {
+  const saveBtn = e.target;
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  const btnClass = saveBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  const taskList = document.querySelector(`#tasks-${id}`);
+  const saveEdit = document.querySelector(`.save-${id}`);
+  const deleteEdit = document.querySelector(`.delete-${id}`);
+  const inputId = document.querySelector(`#complete-${id}`);
+  const editBtn = document.getElementById(`${id}`);
+
+  editBtn.style.display = 'block';
+  inputId.style.outline = 'none';
+
+  existingTodos[id].description = inputId.value;
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+  saveEdit.remove();
+  deleteEdit.remove();
+  taskList.classList.remove('active');
+  inputId.setAttribute('readonly', true);
+};
+
+const completedTodos = (e) => {
+  const checkbox = e.target;
+  let btnId = checkbox.id;
+  btnId = btnId.split('-');
+  const id = parseInt(btnId[1], 10);
+  const inputId = document.querySelector(`#complete-${id}`);
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (checkbox.checked) {
+    existingTodos[id].completed = true;
+    inputId.style.textDecoration = 'line-through';
+    localStorage.setItem('todos', JSON.stringify(existingTodos));
+  } else {
+    existingTodos[id].completed = false;
+    inputId.style.textDecoration = 'none';
+    localStorage.setItem('todos', JSON.stringify(existingTodos));
+  }
```

Comment:
I love the structure of instructions followed here 👏 , all of the checks for the task id and checking if it's completed is perfect 👍 . I just had an idea that it could be made simpler following a KISS methodology instead of having nested if checks and having an else statement that could be completely avoided. you could do something like the given below.
   ```
 if (checkbox.checked) {
existingTodos[id].completed ? false : true;
                localStorage.setItem('todos', JSON.stringify(existingTodos));
              } 
 ```
This makes the check simpler with lesser code than the original and it still works the same way. for example, if item. completed is true it is checked and set as false, if it's false it is checked and set as true.